### PR TITLE
#934 Refactored EndsDaemon to use ShellCommand

### PR DESCRIPTION
**Pull request by original-brownbear**
This addresses #934:
* Created new ShellCommand class to encapsulated running a command in a shell
* Used ShellCommand to make EndsDaemon use fewer objects and remove duplicate
logic for sanitizing and running commands in EndsDaemon

### CODE REVIEW

**Comment by alex-palevsky** [comment]
@original-brownbear I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by alex-palevsky** [comment]
@pinaf help us to review this

File: src/main/java/com/rultor/agents/daemons/ShellCommand.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import com.jcabi.ssh.SSH;
+import com.jcabi.ssh.Shell;
+import java.io.IOException;
+
+/**
+ * Command to run in a given shell and working directory.
+ *
+ * @author Armin Braun (me@obrown.io)
+ * @version $Id$
+ * @since 2.0
```
Comment (pinaf) [inline src/main/java/com/rultor/agents/daemons/ShellCommand.java]: @original-brownbear 1.62

**Comment by pinaf** [comment]
@original-brownbear 1 minor comment

**Comment by original-brownbear** [comment]
@pinaf dealt with it :)

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by original-brownbear** [comment]
@alex-palevsky this can be closed given that it is now merged right ?

**Comment by pinaf** [comment]
@yegor256 @original-brownbear It should have been closed automatically by github after merge. Weird.

**Comment by pinaf** [comment]
@rultor deploy

**Comment by original-brownbear** [comment]
@pinaf I think this is the case discussed here https://github.com/yegor256/rultor/pull/1002.
The MR only closes when the merge could have been outright fast forwarded and the rebase Rultor does would be without effect. If there were actually new commits in between that needed to be rebased upon => hashes change => Github doesn't recognize the merge .

**Comment by alex-palevsky** [comment]
> @alex-palevsky this can be closed given that it is now merged right ?

@original-brownbear just close it yourself please

**Comment by original-brownbear** [comment]
@alex-palevsky ok :)

**Comment by alex-palevsky** [comment]
@pinaf Thanks so much! Your account was topped up for **20 mins** (transaction ID is `AP-4CC02639S5753930T`, the task took 172 hours and 36 mins); extra minutes for review comments (c=5); +20 added to your rating, current score is: [+6503](http://www.netbout.com/b/35128?open=rating)

**Pull request closed without merging**